src/core/message-reply.ts

```diff
@@ -40,4 +40,9 @@ export type UnionMessageReply = GenericMessageReply & {
    * Mutually exclusive with `record`.
    */
   cursor?: PaginationCursor;
+
+  /**
+   * A subscription object if a subscription was requested.
+   */
+  subscription?: GenericMessageSubscription;
```

Comment:
For my understanding: why can't there be just one interface as opposed to having both `EventSubscription` and `GenericMessageSubscription`?